[libxslt] Update to 1.1.34

pastle-ptc opened the pull request:
Fixes #16616
I've created/updated the patch files and configuration files for the port of libxslt at version 1.1.34.  The patch files are an amalgamation of the previous patch files that were in 1.1.33 and the required, additional changes for 1.1.34.

pastle-ptc committed:
Updated to point at 1.1.34 tagged version of libxslt, merged the patch differences and added the required, additional patches to get it building.

pastle-ptc committed:
Added missing version info using "vcpkg x-add-version".

pastle-ptc committed:
Merge branch 'libxslt-1.1.34' of github.com:pastle-ptc/vcpkg into libxslt-1.1.34

NancyLi1013 commented on ports/libxslt/CONTROL:
```diff
@@ -1,6 +1,6 @@
-Version: 1.1.33
-Port-Version: 9
+Version: 1.1.34
+Port-Version: 10
```
```suggestion
```
Since this is a new version update, this field is no longer needed anymore. Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md#port-version

NancyLi1013 commented:
Hi @pastle-ptc 

Thanks for your PR.
Seems like you make a new patch to update changes. Actually, we prefer to patch based on the original one if you fix the same issue.

So could you please update the patches in this PR?

NancyLi1013 commented:
The failures on `x64-windows-static` and  `x64-windows-static-md`:

```
link.exe /nologo /LIBPATH:bin.msvc /LIBPATH:D:\packages\libxslt_x64-windows-static\debug\lib /DEBUG  /DLL  /VERSION:1.1  /IMPLIB:bin.msvc\libxslt.lib /OUT:bin.msvc\libxslt.dll  int.xslt.msvc\attributes.obj int.xslt.msvc\documents.obj int.xslt.msvc\extensions.obj int.xslt.msvc\extra.obj int.xslt.msvc\functions.obj int.xslt.msvc\imports.obj int.xslt.msvc\keys.obj int.xslt.msvc\namespaces.obj int.xslt.msvc\numbers.obj int.xslt.msvc\pattern.obj int.xslt.msvc\preproc.obj int.xslt.msvc\security.obj int.xslt.msvc\templates.obj int.xslt.msvc\transform.obj int.xslt.msvc\variables.obj int.xslt.msvc\xslt.obj int.xslt.msvc\xsltlocale.obj int.xslt.msvc\xsltutils.obj int.xslt.msvc\attrvt.obj iconv.lib charset.lib zlibd.lib lzmad.lib libxml2.lib
   Creating library bin.msvc\libxslt.lib and object bin.msvc\libxslt.exp
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __WSAFDIsSet referenced in function xmlNanoHTTPConnectAttempt
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_closesocket referenced in function xmlNanoHTTPFreeCtxt
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_closesocket
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_connect referenced in function xmlNanoHTTPConnectAttempt
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_connect
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_ioctlsocket referenced in function xmlNanoHTTPConnectAttempt
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_getsockopt referenced in function xmlNanoHTTPConnectAttempt
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_htonl referenced in function WspiapiLegacyGetAddrInfo
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_htons referenced in function xmlNanoHTTPConnectHost
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_htons
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_inet_addr referenced in function WspiapiParseV4Address
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_inet_ntoa referenced in function WspiapiLegacyGetAddrInfo
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_ntohs referenced in function WspiapiLegacyGetNameInfo
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_recv referenced in function xmlNanoHTTPRecv
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_recv
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_select referenced in function xmlNanoHTTPSend
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_select
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_send referenced in function xmlNanoHTTPSend
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_send
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_socket referenced in function xmlNanoHTTPConnectAttempt
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_socket
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_gethostbyaddr referenced in function WspiapiLegacyGetNameInfo
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_gethostbyname referenced in function WspiapiQueryDNS
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_gethostbyname
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_getservbyport referenced in function WspiapiLegacyGetNameInfo
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_getservbyname referenced in function WspiapiLegacyGetAddrInfo
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_WSAStartup referenced in function xmlNanoHTTPInit
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_WSAStartup
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_WSACleanup referenced in function xmlNanoHTTPCleanup
libxml2.lib(nanoftp.c.obj) : error LNK2001: unresolved external symbol __imp_WSACleanup
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_WSASetLastError referenced in function WspiapiGetAddrInfo
libxml2.lib(nanohttp.c.obj) : error LNK2019: unresolved external symbol __imp_WSAGetLastError referenced in function socket_errno
libxml2.lib(nanoftp.c.obj) : error LNK2019: unresolved external symbol __imp_bind referenced in function xmlNanoFTPGetConnection
libxml2.lib(nanoftp.c.obj) : error LNK2019: unresolved external symbol __imp_getsockname referenced in function xmlNanoFTPGetConnection
libxml2.lib(nanoftp.c.obj) : error LNK2019: unresolved external symbol __imp_listen referenced in function xmlNanoFTPGetConnection
bin.msvc\libxslt.dll : fatal error LNK1120: 25 unresolved externals
```

pastle-ptc committed:
Update ports/libxslt/CONTROL

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

pastle-ptc committed:
Re-added the version with the port-version fix in.

pastle-ptc committed:
Fix for x64-windows-static builds.

pastle-ptc committed:
Another re-add of the version since the git tree hash is different.

pastle-ptc commented:
> Hi @pastle-ptc
> 
> Thanks for your PR.
> Seems like you make a new patch to update changes. Actually, we prefer to patch based on the original one if you fix the same issue.
> 
> So could you please update the patches in this PR?

I'm not sure I understand your request...  I merged 0001 patch and 0002 patch, along with the changes required for 1.1.34 of libxslt into a single update to 0001 patch.  What do you want me to do instead?

NancyLi1013 commented:
I mean you can update the changes based on the original patches such as 0001-Fix-makefile.patch and 0002-Fix-lzma.patch.

I noticed that you renamed 0002-Fix-lzma.patch as 0002-Fix-configure.patch. In fact, we suggest to use the existed patch name since it is more clear to see the differences between the two versions.

pastle-ptc commented:
Ah, I think I understand a bit better now.

So originally, for 1.1.33, there were two patch files:

0001-Fix-makefile.patch > contained updates to Makefile.msvc
0002-Fix-lzma.patch > contained updates to Makefile.msvc and configure.js

Neither of these patches were working since Makefile.msvc had been modified in 1.1.34 enough that the patches couldn't match.  So, to get everything working, I took the Makefile.msvc from the build tree for 1.1.33, added in the differences from 1.1.34 by hand and then generated a new patch by differencing the Makefile.msvc in 1.1.34 with my "finished" one I'd done by hand.  This I used to create a new version of 0001-Fix-makefile.patch.  

That just left the configure.js bits of the patch to put back in but that file had changed in 1.1.34 too, so I did a similar process for that file too.  Since it wasn't just a lzma patch now, the file name seemed misleading, so I changed it to make it clear the file it was patching instead.  I guess I could have put both files into a single, new patch file, but I'm new to patches and kept it simple with a file's worth of diff's in each file.

To update the files individually, I'd have had to have manually updated the Makefile.msvc three times to try to simulate what the original two patches did, along with a new patch file to make the additional changes.  Is this what you'd prefer I do?  I was concerned this would leave it quite open to introducing errors or missing something, whereas just differencing the already patched version was much simpler and (I thought) safer.

NancyLi1013 commented:
LGTM now, thanks for your update @pastle-ptc.

ras0219-msft commented on ports/libxslt/0002-Fix-configure.patch:
```diff
@@ -0,0 +1,118 @@
++var withLzma = true;
+ var withCrypto = true;
+ var withModules = false;
++var withProfiler = true;
```
I don't see what this profiler argument is being used for -- could you explain why this needs to be patched in?

pastle-ptc replied:
The withProfiler bits came from a commit that was pushed into the original GNOME repository to fix an error:

https://github.com/GNOME/libxslt/commit/e2584eed1c84c18f16e42188c30d2c3d8e3e8853#diff-c89fc6e6e59d78485a235e28fa5edba702ed057fe8d9f97fa5e8805c574d67f5

The comment read:

> Without this the generated xsltconfig.h will not be complete as there will be a configuration variable that is left in the header, breaking builds.  This will allow one to enable or disable profiler support in Windows builds, and the default is to enable this.

ras0219-msft committed:
[libxslt] Reorganize patches

ras0219-msft reviewed (approved):
Thanks! I've reorganized the patch files to make that clearer and make it easier when we update next.

pastle-ptc commented:
@ras0219-msft it appears that the changes to the patch are causing two test failures that weren't there before.  Who's supposed to fix them?

NancyLi1013 commented:
/azp run

NancyLi1013 commented:
All CI checks have passed now. So there is no need to fix anything.

pastle-ptc commented:
@NancyLi1013 that's great news - thanks!
